From: Nathan Pratta Teodosio <nathan.teodosio@canonical.com>
Date: Mon, 16 Sep 2024 11:19:04 +0200
Subject: Fix Chromium's progressive web application tab and dock grouping.

Bug-Ubuntu: https://launchpad.net/bugs/2007652
Forwarded: not-needed
---
 src/shell-window-tracker.c | 32 ++++++++++++++++++++++++++++++--
 1 file changed, 30 insertions(+), 2 deletions(-)

diff --git a/src/shell-window-tracker.c b/src/shell-window-tracker.c
index 8c3de6f..349495c 100644
--- a/src/shell-window-tracker.c
+++ b/src/shell-window-tracker.c
@@ -153,11 +153,21 @@ get_app_from_window_wmclass (MetaWindow  *window)
   const char *wm_instance;
   const char *sandbox_id;
   g_autofree char *app_prefix = NULL;
+  gboolean is_sandboxed_chromium;
 
   appsys = shell_app_system_get_default ();
 
   sandbox_id = meta_window_get_sandboxed_app_id (window);
-  if (sandbox_id)
+
+  /* Progressive web application tab and dock grouping needs
+   * to be excepted from the special sandboxed logic to work properly
+   * in the Chromium snap (LP:2007652).
+   */
+  is_sandboxed_chromium = sandbox_id &&
+                          g_str_has_prefix (sandbox_id, "chromium_") &&
+                          g_str_has_suffix (sandbox_id, "_chromium");
+
+  if (sandbox_id && !is_sandboxed_chromium)
     app_prefix = g_strdup_printf ("%s.", sandbox_id);
 
   /* Notes on the heuristics used here:
@@ -199,7 +209,25 @@ get_app_from_window_wmclass (MetaWindow  *window)
   wm_instance = meta_window_get_wm_class_instance (window);
   app = shell_app_system_lookup_startup_wmclass (appsys, wm_instance);
   if (app != NULL && check_app_id_prefix (app, app_prefix))
-    return g_object_ref (app);
+    {
+      if (is_sandboxed_chromium)
+        {
+          GDesktopAppInfo *app_info = shell_app_get_app_info (app);
+          const gchar *exe = NULL;
+
+          if (!G_IS_APP_INFO (app_info))
+            return NULL;
+
+          if (!(exe = g_app_info_get_executable (G_APP_INFO (app_info))))
+            return NULL;
+
+          if (g_strcmp0 (exe, "/snap/bin/chromium") != 0 &&
+              !g_str_has_prefix (exe, "/snap/bin/chromium_"))
+            return NULL;
+        }
+
+      return g_object_ref (app);
+    }
 
   /* then try a match from WM_CLASS to StartupWMClass */
   wm_class = meta_window_get_wm_class (window);
