From 5ec0ad4e9cf8a88e3a715fdef4bf1f227b274d2d Mon Sep 17 00:00:00 2001
From: Marc-Antoine Perennou <Marc-Antoine@Perennou.com>
Date: Tue, 12 Sep 2017 12:23:46 +0200
Subject: [PATCH 4/5] StEntry: fix crash when hint actor is NULL

Bug: https://bugzilla.gnome.org/show_bug.cgi?id=787580
Signed-off-by: Marc-Antoine Perennou <Marc-Antoine@Perennou.com>
Origin: upstream, 3.27.0, commit:5ec0ad4e9cf8a88e3a715fdef4bf1f227b274d2d
---
 src/st/st-entry.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/st/st-entry.c b/src/st/st-entry.c
index fc9ffbffd..18e6a257d 100644
--- a/src/st/st-entry.c
+++ b/src/st/st-entry.c
@@ -1106,7 +1106,7 @@ st_entry_get_text (StEntry *entry)
   g_return_val_if_fail (ST_IS_ENTRY (entry), NULL);
 
   priv = st_entry_get_instance_private (entry);
-  if (clutter_actor_is_visible (priv->hint_actor))
+  if (priv->hint_actor != NULL && clutter_actor_is_visible (priv->hint_actor))
     return "";
   else
     return clutter_text_get_text (CLUTTER_TEXT (priv->entry));
-- 
2.14.1

