From: Michael Biebl <biebl@debian.org>
Description: Don't mix intltool and gettext macros.
Forwarded: no

diff --git a/configure.ac b/configure.ac
index 13cd42b..cd579e2 100644
--- a/configure.ac
+++ b/configure.ac
@@ -26,9 +26,6 @@ LT_INIT([disable-static])
 # i18n
 IT_PROG_INTLTOOL([0.40])
 
-AM_GNU_GETTEXT([external])
-AM_GNU_GETTEXT_VERSION([0.17])
-
 GETTEXT_PACKAGE=gnome-shell
 AC_SUBST(GETTEXT_PACKAGE)
 AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE, "$GETTEXT_PACKAGE",
