From: Rico Tzschichholz <ricotz@ubuntu.com>
Description: Remove hard-dep on GdmGreeter introspection.
Forwarded: no

Index: gnome-shell-3.5.4/js/ui/userMenu.js
===================================================================
--- gnome-shell-3.5.4.orig/js/ui/userMenu.js	2012-07-17 11:53:09.000000000 -0400
+++ gnome-shell-3.5.4/js/ui/userMenu.js	2012-07-17 12:42:27.833752552 -0400
@@ -1,7 +1,6 @@
 // -*- mode: js; js-indent-level: 4; indent-tabs-mode: nil -*-
 
 const AccountsService = imports.gi.AccountsService;
-const Gdm = imports.gi.Gdm;
 const Gio = imports.gi.Gio;
 const GLib = imports.gi.GLib;
 const Lang = imports.lang;
@@ -562,7 +561,7 @@
     _updateSwitchUser: function() {
         let allowSwitch = !this._lockdownSettings.get_boolean(DISABLE_USER_SWITCH_KEY);
         let multiUser = this._userManager.can_switch() && this._userManager.has_multiple_users;
-        let multiSession = Gdm.get_session_ids().length > 1;
+        let multiSession = true;
 
         this._loginScreenItem.label.set_text(multiUser ? _("Switch User")
                                                        : _("Switch Session"));
@@ -572,7 +571,7 @@
     _updateLogout: function() {
         let allowLogout = !this._lockdownSettings.get_boolean(DISABLE_LOG_OUT_KEY);
         let multiUser = this._userManager.has_multiple_users;
-        let multiSession = Gdm.get_session_ids().length > 1;
+        let multiSession = true;
 
         this._logoutItem.actor.visible = allowLogout && (multiUser || multiSession);
     },
