Index: gnome-shell-3.4.1/js/ui/userMenu.js
===================================================================
--- gnome-shell-3.4.1.orig/js/ui/userMenu.js	2012-03-19 16:56:49.000000000 +0100
+++ gnome-shell-3.4.1/js/ui/userMenu.js	2012-05-21 02:26:07.307196419 +0200
@@ -636,10 +636,6 @@
         item = new PopupMenu.PopupSeparatorMenuItem();
         this.menu.addMenuItem(item);
 
-        item = new PopupMenu.PopupMenuItem(_("Online Accounts"));
-        item.connect('activate', Lang.bind(this, this._onOnlineAccountsActivate));
-        this.menu.addMenuItem(item);
-
         item = new PopupMenu.PopupMenuItem(_("System Settings"));
         item.connect('activate', Lang.bind(this, this._onPreferencesActivate));
         this.menu.addMenuItem(item);
@@ -698,12 +694,6 @@
         app.activate();
     },
 
-    _onOnlineAccountsActivate: function() {
-        Main.overview.hide();
-        let app = Shell.AppSystem.get_default().lookup_setting('gnome-online-accounts-panel.desktop');
-        app.activate(-1);
-    },
-
     _onPreferencesActivate: function() {
         Main.overview.hide();
         let app = Shell.AppSystem.get_default().lookup_app('gnome-control-center.desktop');
