From 8845a2170c01f83ac23b54ef6f6389b94b142f78 Mon Sep 17 00:00:00 2001
From: Clemens Buchacher <drizzd@aon.at>
Date: Sun, 01 Jul 2012 16:48:48 +0000
Subject: networkAgent: use absolute path if configured

Signed-off-by: Clemens Buchacher <drizzd@aon.at>

https://bugzilla.gnome.org/show_bug.cgi?id=679212
---
diff --git a/js/ui/networkAgent.js b/js/ui/networkAgent.js
index cc44fa2..c9473a7 100644
--- a/js/ui/networkAgent.js
+++ b/js/ui/networkAgent.js
@@ -683,7 +683,10 @@ const NetworkAgent = new Lang.Class({
                     try {
                         externalUIMode = keyfile.get_boolean('GNOME', 'supports-external-ui-mode');
                     } catch(e) { } // ignore errors if key does not exist
-                    let path = GLib.build_filenamev([Config.LIBEXECDIR, binary]);
+                    let path = binary;
+                    if (!GLib.path_is_absolute(path)) {
+                        path = GLib.build_filenamev([Config.LIBEXECDIR, path]);
+                    }
 
                     if (GLib.file_test(path, GLib.FileTest.IS_EXECUTABLE))
                         this._vpnBinaries[service] = { fileName: path, externalUIMode: externalUIMode };
--
cgit v0.9.0.2
