Description: Use Debian specific NM VPN helpers paths
Author: Laurent Bigonville <bigon@debian.org>
Forwarded: not-needed

--- a/js/ui/components/networkAgent.js
+++ b/js/ui/components/networkAgent.js
@@ -673,7 +673,7 @@
                     } catch(e) { } // ignore errors if key does not exist
                     let path = binary;
                     if (!GLib.path_is_absolute(path)) {
-                        path = GLib.build_filenamev([Config.LIBEXECDIR, path]);
+                        path = GLib.build_filenamev(['/usr/lib/NetworkManager', path]);
                     }
 
                     if (GLib.file_test(path, GLib.FileTest.IS_EXECUTABLE))
