From: Laurent Bigonville <bigon@debian.org>
Date: Sun, 7 Jan 2018 17:15:51 +0000
Subject: Use Debian specific NM VPN helpers paths

Forwarded: not-needed
---
 js/ui/components/networkAgent.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/js/ui/components/networkAgent.js b/js/ui/components/networkAgent.js
index fd59c11..b6a3456 100644
--- a/js/ui/components/networkAgent.js
+++ b/js/ui/components/networkAgent.js
@@ -772,7 +772,7 @@ var NetworkAgent = new Lang.Class({
 
                     let path = binary;
                     if (!GLib.path_is_absolute(path)) {
-                        path = GLib.build_filenamev([Config.LIBEXECDIR, path]);
+                        path = GLib.build_filenamev(['/usr/lib/NetworkManager', path]);
                     }
 
                     if (GLib.file_test(path, GLib.FileTest.IS_EXECUTABLE)) {
