From e6f5e21b5d416746dd19628791bc93c43dc382f1 Mon Sep 17 00:00:00 2001
From: Matthias Clasen <mclasen@redhat.com>
Date: Fri, 15 Jun 2012 21:54:04 +0000
Subject: Enable the Screen Reader menu item

https://bugzilla.gnome.org/show_bug.cgi?id=663256
---
diff --git a/js/ui/status/accessibility.js b/js/ui/status/accessibility.js
index 42d3a3c..20c54c6 100644
--- a/js/ui/status/accessibility.js
+++ b/js/ui/status/accessibility.js
@@ -56,9 +56,9 @@ const ATIndicator = new Lang.Class({
         let textZoom = this._buildFontItem();
         this.menu.addMenuItem(textZoom);
 
-//        let screenReader = this._buildItem(_("Screen Reader"), APPLICATIONS_SCHEMA,
-//                                                               'screen-reader-enabled');
-//        this.menu.addMenuItem(screenReader);
+        let screenReader = this._buildItem(_("Screen Reader"), APPLICATIONS_SCHEMA,
+                                                               'screen-reader-enabled');
+        this.menu.addMenuItem(screenReader);
 
         let screenKeyboard = this._buildItem(_("Screen Keyboard"), APPLICATIONS_SCHEMA,
                                                                    'screen-keyboard-enabled');
--
cgit v0.9.0.2
