--- a/src/shell-global.c
+++ b/src/shell-global.c
@@ -476,12 +476,12 @@
                                                         G_PARAM_READABLE));
 }
 
-/**•
- * _shell_global_init: (skip)•
+/*
+ * _shell_global_init: (skip)
  * @first_property_name: the name of the first property
  * @...: the value of the first property, followed optionally by more
  *  name/value pairs, followed by %NULL
- *•
+ *
  * Initializes the shell global singleton with the construction-time
  * properties.
  *
