--- a/js/ui/status/bluetooth.js
+++ b/js/ui/status/bluetooth.js
@@ -36,11 +36,11 @@
         this._applet.connect('notify::killswitch-state', Lang.bind(this, this._updateKillswitch));
         this._killswitch.connect('toggled', Lang.bind(this, function() {
             let current_state = this._applet.killswitch_state;
-            if (current_state != GnomeBluetoothApplet.KillswitchState.HARD_BLOCKED &&
-                current_state != GnomeBluetoothApplet.KillswitchState.NO_ADAPTER) {
+            if (current_state != GnomeBluetoothApplet.BluetoothKillswitchState.HARD_BLOCKED &&
+                current_state != GnomeBluetoothApplet.BluetoothKillswitchState.NO_ADAPTER) {
                 this._applet.killswitch_state = this._killswitch.state ?
-                    GnomeBluetoothApplet.KillswitchState.UNBLOCKED:
-                    GnomeBluetoothApplet.KillswitchState.SOFT_BLOCKED;
+                    GnomeBluetoothApplet.BluetoothKillswitchState.UNBLOCKED:
+                    GnomeBluetoothApplet.BluetoothKillswitchState.SOFT_BLOCKED;
             } else
                 this._killswitch.setToggleState(false);
         }));
@@ -94,10 +94,10 @@
 
     _updateKillswitch: function() {
         let current_state = this._applet.killswitch_state;
-        let on = current_state == GnomeBluetoothApplet.KillswitchState.UNBLOCKED;
-        let has_adapter = current_state != GnomeBluetoothApplet.KillswitchState.NO_ADAPTER;
-        let can_toggle = current_state != GnomeBluetoothApplet.KillswitchState.NO_ADAPTER &&
-                         current_state != GnomeBluetoothApplet.KillswitchState.HARD_BLOCKED;
+        let on = current_state == GnomeBluetoothApplet.BluetoothKillswitchState.UNBLOCKED;
+        let has_adapter = current_state != GnomeBluetoothApplet.BluetoothKillswitchState.NO_ADAPTER;
+        let can_toggle = current_state != GnomeBluetoothApplet.BluetoothKillswitchState.NO_ADAPTER &&
+                         current_state != GnomeBluetoothApplet.BluetoothKillswitchState.HARD_BLOCKED;
 
         this._killswitch.setToggleState(on);
         if (can_toggle)
