From eb759cf22fc6afd9a0d0afb5bfbc34851413ca12 Mon Sep 17 00:00:00 2001
From: Colin Walters <walters@verbum.org>
Date: Tue, 26 Jun 2012 18:00:35 +0000
Subject: dateMenu: Fix regression that caused no date to be displayed

Commit ef0aa65774af02ef86d6b131b336c3914c6daf97 broke the date
display; bring it back.
---
diff --git a/js/ui/dateMenu.js b/js/ui/dateMenu.js
index 878cd35..5e5b3af 100644
--- a/js/ui/dateMenu.js
+++ b/js/ui/dateMenu.js
@@ -155,6 +155,12 @@ const DateMenuButton = new Lang.Class({
 
     _updateClockAndDate: function() {
         this._clockDisplay.set_text(this._clock.clock);
+        /* Translators: This is the date format to use when the calendar popup is
+         * shown - it is shown just below the time in the shell (e.g. "Tue 9:29 AM").
+         */
+        let dateFormat = _("%A %B %e, %Y");
+        let displayDate = new Date();
+        this._date.set_text(displayDate.toLocaleFormat(dateFormat));
     },
 
     _onOpenCalendarActivate: function() {
