From: =?utf-8?q?Florian_M=C3=BCllner?= <fmuellner@gnome.org>
Date: Mon, 9 Jul 2018 13:31:26 +0200
Subject: calendar: chain up to parent on _onDestroy

---
 js/ui/calendar.js | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/js/ui/calendar.js b/js/ui/calendar.js
index 990cac2..815c9f9 100644
--- a/js/ui/calendar.js
+++ b/js/ui/calendar.js
@@ -803,6 +803,8 @@ var NotificationMessage = new Lang.Class({
     },
 
     _onDestroy() {
+        this.parent();
+
         if (this._updatedId)
             this.notification.disconnect(this._updatedId);
         this._updatedId = 0;
