From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Tue, 2 Feb 2021 16:40:24 +0100
Subject: gdm: Include the failed service name when in reporting errors

Origin: https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/1622
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/gnome-shell/+bug/1915066
---
 js/gdm/util.js | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/js/gdm/util.js b/js/gdm/util.js
index b40bc23..8c6c678 100644
--- a/js/gdm/util.js
+++ b/js/gdm/util.js
@@ -459,8 +459,8 @@ var ShellUserVerifier = class {
                 return;
             }
             this._reportInitError(this._userName
-                ? 'Failed to start verification for user'
-                : 'Failed to start verification', e);
+                ? 'Failed to start %s verification for user'.format(serviceName)
+                : 'Failed to start %s verification'.format(serviceName), e);
             return;
         }
         this._hold.release();
