--- a/js/ui/appDisplay.js
+++ b/js/ui/appDisplay.js
@@ -2497,7 +2497,10 @@
                 this._appendSeparator();
             }
 
+            let vendor = Shell.util_get_gl_vendor();
             if (discreteGpuAvailable &&
+                vendor != "NVIDIA Corporation" &&
+                vendor != "nouveau" &&
                 this._source.app.state == Shell.AppState.STOPPED) {
                 this._onDiscreteGpuMenuItem = this._appendMenuItem(_("Launch using Dedicated Graphics Card"));
                 this._onDiscreteGpuMenuItem.connect('activate', () => {
--- a/src/shell-util.c
+++ b/src/shell-util.c
@@ -379,8 +379,8 @@
 
 typedef const gchar *(*ShellGLGetString) (GLenum);
 
-static const gchar *
-get_gl_vendor (void)
+const gchar *
+shell_util_get_gl_vendor (void)
 {
   static const gchar *vendor = NULL;
 
@@ -401,7 +401,7 @@
   if (!clutter_check_windowing_backend (CLUTTER_WINDOWING_X11))
     return FALSE;
 
-  if (g_strcmp0 (get_gl_vendor (), "NVIDIA Corporation") == 0)
+  if (g_strcmp0 (shell_util_get_gl_vendor (), "NVIDIA Corporation") == 0)
     return TRUE;
 
   return FALSE;
--- a/src/shell-util.h
+++ b/src/shell-util.h
@@ -44,6 +44,8 @@
                                                int                height,
                                                int                rowstride);
 
+const gchar *shell_util_get_gl_vendor (void);
+
 gboolean shell_util_need_background_refresh (void);
 
 ClutterContent * shell_util_get_content_for_window_actor (MetaWindowActor *window_actor,
