Index: gnome-shell/js/ui/background.js
===================================================================
--- gnome-shell.orig/js/ui/background.js
+++ gnome-shell/js/ui/background.js
@@ -606,7 +606,7 @@ var BackgroundSource = class BackgroundS
                 const colorScheme = this._interfaceSettings.get_enum('color-scheme');
                 const uri = this._settings.get_string(
                     colorScheme === GDesktopEnums.ColorScheme.PREFER_DARK
-                        ? PICTURE_URI_DARK_KEY
+                        ? PICTURE_URI_KEY /* TODO: PICTURE_URI_DARK_KEY */
                         : PICTURE_URI_KEY);
 
                 file = Gio.File.new_for_commandline_arg(uri);
