Index: gnome-shell/js/ui/notificationDaemon.js
===================================================================
--- gnome-shell.orig/js/ui/notificationDaemon.js
+++ gnome-shell/js/ui/notificationDaemon.js
@@ -34,11 +34,6 @@ var FdoNotificationDaemon = class FdoNot
         this._notifications = {};
 
         this._nextNotificationId = 1;
-
-        Shell.WindowTracker.get_default().connect('notify::focus-app',
-            this._onFocusAppChanged.bind(this));
-        Main.overview.connect('hidden',
-            this._onFocusAppChanged.bind(this));
     }
 
     _imageForNotificationData(hints) {
@@ -326,20 +321,6 @@ var FdoNotificationDaemon = class FdoNot
         ];
     }
 
-    _onFocusAppChanged() {
-        let tracker = Shell.WindowTracker.get_default();
-        if (!tracker.focus_app)
-            return;
-
-        for (let i = 0; i < this._sources.length; i++) {
-            let source = this._sources[i];
-            if (source.app == tracker.focus_app) {
-                source.destroyNonResidentNotifications();
-                return;
-            }
-        }
-    }
-
     _emitNotificationClosed(id, reason) {
         this._dbusImpl.emit_signal('NotificationClosed',
                                    GLib.Variant.new('(uu)', [id, reason]));
