Index: gnome-shell/js/ui/main.js
===================================================================
--- gnome-shell.orig/js/ui/main.js
+++ gnome-shell/js/ui/main.js
@@ -418,6 +418,10 @@ function _getDefaultStylesheet() {
     if (St.Settings.get().high_contrast)
         stylesheet = _getStylesheet(name.replace('.css', '-high-contrast.css'));
 
+    // Look for -dark theme if GTK theme variant is dark
+    if (St.Settings.get().gtk_theme_variant == 'dark')
+        stylesheet = _getStylesheet(name.replace('.css', '-dark.css'));
+
     if (stylesheet == null) {
         const settings = St.Settings.get();
 
