Index: gnome-shell/js/ui/layout.js
===================================================================
--- gnome-shell.orig/js/ui/layout.js
+++ gnome-shell/js/ui/layout.js
@@ -352,8 +352,7 @@ var LayoutManager = GObject.registerClas
             this.bottomMonitor = this.monitors[this.bottomIndex];
 
             if (this._pendingLoadBackground) {
-                this._loadBackground();
-                this._pendingLoadBackground = false;
+                this._pendingLoadBackground = this._loadBackground() ? true : false;
             }
         } else {
             this.primaryMonitor = null;
@@ -616,13 +615,19 @@ var LayoutManager = GObject.registerClas
             // This helps to prevent us from running the animation
             // when the system is bogged down
             const id = GLib.idle_add(GLib.PRIORITY_LOW, () => {
-                this._systemBackground.show();
-                global.stage.show();
-                this._prepareStartupAnimation();
-                return GLib.SOURCE_REMOVE;
+                if (this.primaryMonitor) {
+                    this._systemBackground.show();
+                    global.stage.show();
+                    this._prepareStartupAnimation();
+                    return GLib.SOURCE_REMOVE;
+                } else {
+                    return GLib.SOURCE_CONTINUE;
+                }
+
             });
             GLib.Source.set_name_by_id(id, '[gnome-shell] Startup Animation');
         });
+        return true;
     }
 
     // Startup Animations
